File: 助教課/20231024/Q1.c
#include <stdio.h>
#include <math.h>

void decimalToBinary(int num) { 
	int k,x;
	long long m = 0;
	
    if (num == 0) {
        printf("0");
        return;
    }
    
    if(num>0){
    
		for(k=0;pow(2,k)<=num;k++){
			x=k;
		}
    	
    	for(x;x>=0;x--){
    		
			if( (num - pow(2,x))>=0 ){
    			num = num - pow(2,x);
    			m = m + pow(10,x);
			}
		}
    	printf("%lld",m);
	}
}

int main() {
    int num;
    
    printf("Enter a decimal number: ");
    scanf("%d", &num);

    decimalToBinary(num);

    return 0;
}

File: 助教課/20231024/Q1.py
m = 0
num = int(input("Enter a decimal number: "))

if num == 0:
    print("0")

if num > 0:
    
    k = 0
    while 2**k <= num:
        k = k + 1

    x = k - 1 #最大到2的x次方

    while x >= 0:

        if num - 2**x >=0:
            num = num - 2**x
            m = m + 10**x
        x = x - 1

    print(m)
